sql: TestTemporarySchemaDropDatabase failed

[(sql).TestTemporarySchemaDropDatabase failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2275673&tab=buildLog) on [master@2fd9d953c6ae435c20fdb1acce1eb352f37f6234](https://github.com/cockroachdb/cockroach/commits/2fd9d953c6ae435c20fdb1acce1eb352f37f6234):

```
=== RUN   TestTemporarySchemaDropDatabase
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTemporarySchemaDropDatabase819380436
--- FAIL: TestTemporarySchemaDropDatabase (36.21s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTemporarySchemaDropDatabase819380436
    test_log_scope.go:63: use -show-logs to present logs inline
    temporary_schema_test.go:335: 
        	Error Trace:	temporary_schema_test.go:335
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestTemporarySchemaDropDatabase

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTemporarySchemaDropDatabase PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTemporarySchemaDropDatabase.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTemporarySchemaDropDatabase failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2278904&tab=buildLog) on [master@36609f49a36726e415157b235c74741c917de60f](https://github.com/cockroachdb/cockroach/commits/36609f49a36726e415157b235c74741c917de60f):

```
=== RUN   TestTemporarySchemaDropDatabase
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTemporarySchemaDropDatabase251730173
--- FAIL: TestTemporarySchemaDropDatabase (36.78s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTemporarySchemaDropDatabase251730173
    test_log_scope.go:63: use -show-logs to present logs inline
    temporary_schema_test.go:335: 
        	Error Trace:	temporary_schema_test.go:335
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestTemporarySchemaDropDatabase

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTemporarySchemaDropDatabase PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTemporarySchemaDropDatabase.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/temporary_schema_test.go
+++ b/pkg/sql/temporary_schema_test.go
@@ -327,12 +327,17 @@ func TestTemporarySchemaDropDatabase(t *testing.T) {
 		sqlDB := sqlutils.MakeSQLRunner(db)
 		sqlDB.Exec(t, `DROP DATABASE drop_me CASCADE`)
 
-		var tempObjectCount int
-		sqlDB.QueryRow(
-			t,
-			`SELECT count(1) FROM system.namespace WHERE name LIKE 'pg_temp%' OR name IN ('t', 't2')`,
-		).Scan(&tempObjectCount)
-		assert.Equal(t, 0, tempObjectCount)
+		testutils.SucceedsSoon(t, func() error {
+			var tempObjectCount int
+			sqlDB.QueryRow(
+				t,
+				`SELECT count(1) FROM system.namespace WHERE name LIKE 'pg_temp%' OR name IN ('t', 't2')`,
+			).Scan(&tempObjectCount)
+			if tempObjectCount == 0 {
+				return nil
+			}
+			return errors.AssertionFailedf("expected count 0, got %d", tempObjectCount)
+		})
 	}
 }
 

